### PR TITLE
Adding responseTimeMs to actuator health indicators #9238

### PR DESCRIPTION
**Pull request by strattonw**
Adds responseTimeMs detail to all health indicators as well as the
aggregate for an overall response time.

Fixes gh-9238

### CODE REVIEW

File: spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/CompositeHealthIndicator.java [hunk 1]
```diff
@@ -63,11 +64,39 @@ public void addHealthIndicator(String name, HealthIndicator indicator) {
...
+
+		Health aggregateHealths = this.healthAggregator.aggregate(healths);
+
+		stopWatch.stop();
+
+		return addResponseTimeMsToHealth(aggregateHealths, stopWatch);
+	}
+
+	private Health healthIndicatorWithResponseTime(HealthIndicator healthIndicator) {
+		StopWatch stopWatch = new StopWatch();
```
Comment (igreenfield) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/CompositeHealthIndicator.java]: Why using StopWatch? this create many short live objects for not resean.

Comment (strattonw) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/CompositeHealthIndicator.java]: You're right with the short lived objects, I've updated to make it a bit clearer for utilizing StopWatch and reducing the amount of objects created

**Comment by philwebb** [comment]
Nice! Thanks. I wonder if we might also want to offer a property to opt in/out of this?

**Comment by wilkinsona** [comment]
I wonder if this might be better implemented using the decorator pattern? I also think we should consider what complications this might add to offering reactive actuator endpoints. It feels like decorating something that returns `Mono<Health>` would work quite well but I think some exploratory work would be necessary to know if it's the best approach.

**Comment by philwebb** [comment]
Actually, that's a very good point @wilkinsona. We should probably hold off until we have a clearer picture of the reactive actuator endpoints. Regardless, I think this is a really nice feature for us to add and this code definitely gives us a nice head start.

**Comment by philwebb** [comment]
After a little more consideration we've decided that directly recording the response time in the health indicator itself might not be the best approach. Probably a better solution would be to use Micrometer to directly monitor the underlying infrastructure that the heath endpoints are monitoring.

Thanks anyway for the PR @strattonw

**Pull request closed without merging**